software/runtime/arch.ld.c

```diff
@@ -34,19 +34,37 @@ SECTIONS {
   // Hardware register location
   eoc_reg                = 0x40000000;
   wake_up_reg            = 0x40000004;
-  tcdm_start_address_reg = 0x40000008;
-  tcdm_end_address_reg   = 0x4000000C;
-  nr_cores_address_reg   = 0x40000010;
-  ro_cache_enable        = 0x40000014;
-  ro_cache_flush         = 0x40000018;
-  ro_cache_start_0       = 0x4000001C;
-  ro_cache_end_0         = 0x40000020;
-  ro_cache_start_1       = 0x40000024;
-  ro_cache_end_1         = 0x40000028;
-  ro_cache_start_2       = 0x4000002C;
-  ro_cache_end_2         = 0x40000030;
-  ro_cache_start_3       = 0x40000034;
-  ro_cache_end_3         = 0x40000038;
+  wake_up_group_reg      = 0x40000008;
+  tcdm_start_address_reg = 0x4000000C;
+  tcdm_end_address_reg   = 0x40000010;
+  nr_cores_address_reg   = 0x40000014;
+  ro_cache_enable        = 0x40000018;
+  ro_cache_flush         = 0x4000001C;
+  ro_cache_start_0       = 0x40000020;
+  ro_cache_end_0         = 0x40000024;
+  ro_cache_start_1       = 0x40000028;
+  ro_cache_end_1         = 0x4000002C;
+  ro_cache_start_2       = 0x40000030;
+  ro_cache_end_2         = 0x40000034;
+  ro_cache_start_3       = 0x40000038;
+  ro_cache_end_3         = 0x4000003C;
+
+#if ((NUM_CORES == 256) || (NUM_CORES == 16))
+  wake_up_tile_g0_reg = 0x40000040;
+  wake_up_tile_g1_reg = 0x40000044;
+  wake_up_tile_g2_reg = 0x40000048;
+  wake_up_tile_g3_reg = 0x4000004C;
+#elif NUM_CORES == 1024
+  wake_up_tile_g0_reg = 0x40000040;
+  wake_up_tile_g1_reg = 0x40000044;
+  wake_up_tile_g2_reg = 0x40000048;
+  wake_up_tile_g3_reg = 0x4000004C;
+  wake_up_tile_g4_reg = 0x40000050;
+  wake_up_tile_g5_reg = 0x40000054;
+  wake_up_tile_g6_reg = 0x40000058;
+  wake_up_tile_g7_reg = 0x4000005C;
+#endif
```

Comment:
I am afraid of the direction this is going... How expensive would be to make this uniform?